### ISSUE TITLE
Blank screen with no data in list

### ISSUE THREAD

**Issue by devaar100** [commons-app/apps-android-commons#1363]
**Summary:** 
On contribution list fragment and nearby places list fragment when there is no data to populate list view there is just a blank screen
Since the app is about images, aesthetic beauty is integral 

**Expected behavior:** 
A simple and beautiful png/clip art to be displayed in the centre if the list is empty

**Observed behavior:** 
Blank screen

**Comment by ShivaniGawande** [comment]
Can I work on this ? @devaar100

**Comment by devaar100** [comment]
@ShivaniGawande sry but I want to work on this

**Comment by devaar100** [comment]
Suggestions for nearby places list screen
![3e03fff14fad115ed53d97cf720b1982](https://user-images.githubusercontent.com/29582559/37867275-e2f8f392-2fbb-11e8-9f2e-fc229afb6604.jpg)
![ab46f940845c751642aa967b867d1a1e](https://user-images.githubusercontent.com/29582559/37867276-e3457dd4-2fbb-11e8-8be0-e0f2510479ad.jpg)
![boy_with_luggage](https://user-images.githubusercontent.com/29582559/37867277-e38c2720-2fbb-11e8-9305-8929c6788fe9.png)

**Comment by devaar100** [comment]
Suggestions for contribution list fragment
![camera-upload-512](https://user-images.githubusercontent.com/29582559/37867292-049fbb20-2fbc-11e8-89f2-8af15783eee6.png)
![camera-upload](https://user-images.githubusercontent.com/29582559/37867293-04f5acd8-2fbc-11e8-9112-1b217a69c13b.png)

**Comment by misaochan** [comment]
When you say "no data to populate list view", do you mean that the user is offline? In that case, I think the text "No connection, unable to load contributions/nearby places" would be better. There can also be an image to go along with the text, but it should be something that reflects what the text says, IMO.

For the contribution list, if it is empty because the user has not uploaded anything, I think it would be more useful to instead encourage the user to upload photos instead. Again, there can be a picture, but it should reflect the text.

**Comment by neslihanturan** [comment]
These pictures looks different than our apps style. Plus, their license is very important, are they CC0? Should we include attribution?.

**Comment by devaar100** [comment]
@neslihanturan @misaochan if the issue seems to be of importance then I would invest some time to find more CC0 images that go better with app style
Please guide me for the same :)

**Comment by neslihanturan** [comment]
I never needed such images as a user. However, this can be a good idea to improve communication worth user. I agree with @misaochan images should be together with texts. But I think images should be designed for our app, using existing images doesn't seem like a good idea to me. I don't think we have a designer contributor either.

**Comment by veryveryharry** [comment]
Hi, I can take a stab at creating some graphics for the empty states if this is still available.

**Comment by 97balakrishnan** [comment]
I think these image drawable assets provided in android studio by default under apache license 2.0 can be used initially.
![alt link](https://image.ibb.co/mV7eVS/Screenshot_from_2018_03_31_19_39_38.png)

![alt link](https://image.ibb.co/eyVux7/Screenshot_from_2018_03_31_19_36_12.png)

**Comment by veryveryharry** [comment]
@misaochan that works for me.

Just to clarify, are these the cases were talking about for blank screens?
1. No connection when loading nearby places
2. No connection when loading contributions
3. User has not uploaded anything

I had a few ideas for (1), and was trying to make these consistent with mostly the tutorial illustrations in terms of style.  Let me know if it's not too far off.

Mocks:
![nearbyplaces-01](https://user-images.githubusercontent.com/16273340/38170481-6ec22b58-353b-11e8-8768-ac969964c8c6.png)
![nearbyplaces-02](https://user-images.githubusercontent.com/16273340/38170482-6edb3b0c-353b-11e8-9b70-81e0254137f8.png)
![nearbyplaces-03](https://user-images.githubusercontent.com/16273340/38170483-6ef04d94-353b-11e8-906b-ea8aa3b92f29.png)

**Comment by misaochan** [comment]
Yes, for the blank screens. :)

The mocks look great IMO! I agree that they are consistent with the style in the tutorial. I would probably make the images a little bit darker though, as it's hard to see them at first glance.

Will ping @neslihanturan for further thoughts.

**Comment by neslihanturan** [comment]
I agree they should be a little darker. Besides you should test them on dark theme too, thanks @veryveryharry

**Comment by veryveryharry** [comment]
Sure! I made the colors darker and created a set based on the one I thought was most clear:
![asset 7](https://user-images.githubusercontent.com/16273340/38284103-773d7d7a-376e-11e8-8c65-ced40ac7a87e.png)

Mockups:

|   |   |   |
| --- | --- | --- |
| ![asset 1](https://user-images.githubusercontent.com/16273340/38284108-7bec44c8-376e-11e8-8690-0bdb593dbefb.png) | ![asset 2](https://user-images.githubusercontent.com/16273340/38284109-7bffdb96-376e-11e8-8110-9405ca8a6e86.png) | ![asset 3](https://user-images.githubusercontent.com/16273340/38284110-7c14e87e-376e-11e8-84e2-742cdd802af7.png) |
| ![asset 4](https://user-images.githubusercontent.com/16273340/38284111-7c293edc-376e-11e8-8538-3bbedf98c90b.png) | ![asset 5](https://user-images.githubusercontent.com/16273340/38284112-7c3ecf68-376e-11e8-880f-edf82446ebb5.png) | ![asset 6](https://user-images.githubusercontent.com/16273340/38284113-7c6ad1f8-376e-11e8-95fc-6984b0d447b7.png) |

Current SVGs are in a repository under CC0 license [here](https://github.com/veryveryharry/cc0-images).

Let me know if this works.

**Comment by misaochan** [comment]
@neslihanturan I don't think he was asking us to choose - I think the 3 columns correspond to 3 different scenarios - (1) user hasn't uploaded anything yet, (2) user has no internet connection in contribution list, (3) user has no internet connection in Nearby places? @veryveryharry could you clarify this?

Going with that assumption, for the "no uploads yet" scenario, the first column looks fantastic IMO. :) Great way to encourage users.

For the "no internet connection" scenario, I agree that displaying the wireless icon (bottom of your 2nd last post) with the words "No internet connection" would be better.

**Comment by veryveryharry** [comment]
Yes, the three were for the three different scenarios. Sorry if that wasn't clear.
How is this?

| No uploads | No connectivity (For both uploads & places) |
| --- | --- |
| ![uploads](https://user-images.githubusercontent.com/16273340/38401885-47d38732-390d-11e8-9bff-93aef6a24ee6.png) | ![uploads_noconnect](https://user-images.githubusercontent.com/16273340/38401414-793c845c-390a-11e8-8d2b-d8879c61447d.png) |

I was hoping to convey the idea of "contribution" with the _No uploads_ graphic, but seeing it without any explicit mention of that in the text, I'm thinking it loses a little bit of meaning/clarity?

I did have a few other variations for that one that I didn't show
#### Other options for "No Uploads Yet" 
<img width="920" alt="screen shot 2018-04-05 at 10 48 59 pm" src="https://user-images.githubusercontent.com/16273340/38405463-d7d0fd4e-3924-11e8-8a1b-e92efa39b4c4.png">

**Comment by neslihanturan** [comment]
I would use the middle one from **Other Options** for no uploads yet.

**Comment by veryveryharry** [comment]
Vector files updated [here](https://github.com/veryveryharry/cc0-images/tree/master/images)

**Comment by misaochan** [comment]
Yeah, I agree with the middle image for "no uploads yet".
 
However, for the person who implements this... The text:
**No Recent Uploads**
Make a contribution to Wikimedia Commons today!

that was proposed in the earlier mockup seems perfect to me, to go along with the middle image. I would strongly prefer it over "No Uploads Yet".

Thanks for the vector files @veryveryharry ! :) @devaar100 are you still interested in implementing this?

**Comment by nicolas-raoul** [comment]
veryveryharry's mockup below looks like the best design to me, though I would replace the subtitle with `Contribute one of your pictures to Wikimedia Commons by tapping the + button.`

![asset 1](https://user-images.githubusercontent.com/16273340/38284108-7bec44c8-376e-11e8-8690-0bdb593dbefb.png)
